src/App.js

```diff
@@ -1,19 +1,156 @@
 import React from 'react';
+
+import { Button, Alert, Col, Container, Form, Row } from 'react-bootstrap';
+
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import { TodoList } from './components/TodoList';
+
 import './App.css';
 
-import users from './api/users';
+const preparedTodos = todosFromServer.map(
+  item => ({
+    ...item,
+    user: usersFromServer.find(person => (person.id === item.userId)),
+  }),
+);
+
+class App extends React.Component {
+  state = {
+    todos: preparedTodos,
+    isTitleEmpty: false,
+    title: '',
+    newUser: null,
+    selectedUserId: '',
+    isSelectedUserEmpty: false,
+    wasTodoAdded: false,
+  }
+
+  addTodo = (event) => {
+    event.preventDefault();
+
+    const { title, newUser } = this.state;
+
+    if (title.trim() && newUser) {
+      this.setState(state => (
+        {
+          todos: [...state.todos, {
+            user: newUser,
+            userId: newUser.id,
+            id: state.todos[state.todos.length - 1].id + 1,
+            title,
+            completed: false,
+          }],
+          selectedUserId: '',
+          title: '',
+          newUser: null,
+          isTitleEmpty: false,
+          isSelectedUserEmpty: false,
+          wasTodoAdded: true,
+        }
+      ));
+    }
+
+    if (!title.trim()) {
+      this.setState({
+        isTitleEmpty: true,
+        wasTodoAdded: false,
+      });
+    }
+
+    if (!newUser) {
+      this.setState({
+        isSelectedUserEmpty: true,
+        wasTodoAdded: false,
+      });
+    }
+  }
+
+  handleUserChange = (event) => {
+    const userId = event.target.value;
+
+    this.setState(
+      {
+        newUser: usersFromServer.find(user => user.id === +userId),
```

Comment:
```suggestion
        newUser: usersFromServer.find(user => user.id === Number(userId)),
```

Краще вказати прямо дуууууже явно